agent/php_internal_instrument.c

```diff
@@ -796,6 +800,34 @@ static void nr_php_instrument_datastore_operation_call(
   }
 }
 
+/* 
+ * Like nr_php_instrument_datastore_operation_call() but does not
+ * call the underlying internal function
+ */
+static void nr_php_instrument_datastore_operation(
+    nr_datastore_t datastore,
+    const char* operation,
+    nr_datastore_instance_t* instance,
+    bool instance_only,
+    INTERNAL_FUNCTION_PARAMETERS) {
+  nr_segment_t* segment = NULL;
+  nr_segment_datastore_params_t params = {
+      .datastore = {
+          .type = datastore,
+      },
+      .operation = nr_strdup(operation),
+      .instance  = instance,
+      .instance_only = instance_only,
+      .callbacks = {
+          .backtrace = nr_php_backtrace_callback,
+      },
+  };
+
+  segment = nr_segment_start(NRPRG(txn), NULL, NULL);
+  nr_segment_datastore_end(&segment, &params);
```

Comment:
This operation will probably result in an immediate (nanoseconds) segment duration. Is the rationale for this just that we have to send this data via a segment to the backend?

Reply:
Yes, the datastore metric needs to be send to the backend, and the length here isn't entirely important for several reasons. 1) this datastore metric is only an instance metric, and is not associated with any operation. 2) the operation being wrapped, `add_servers`, is not a function that the customer will likely care about timings of

Reply:
It sounds like we're going to be sending incorrect duration timings to the backend for this specific function call. I'm slightly worried that someone out there will care about these timings, but off the top of my head I can't think of a workaround that will generate all the segments we need in this manner without forcing a rewrite of our segment logic.

Reply:
It could call the metric creation function after the internal call and pass in the timing information to override the timings automatically generated by the segment start/end. It's not pretty, but would work

Reply:
Or potentially a small function that can loop over a set of segments that have already had timings set and update `segment->start_time` and `segment->stop_time` to appropriate values? Store `start` and `stop` as variables within the wrapper function then passing them as the values to the custom function?
I don't know if that would have any side-effects...

Reply:
But this does raise the important question: will these metrics -- even with correct timings -- confuse customers? There will be much longer instance metrics associated with all of the different datastore operations of the transaction, and then a few much shorter instance metrics at the start of each transaction that is not clear what they are

Reply:
Torn on this as well- would be good to see how this will be displayed in the UI to try and gauge the "confusion" factor

Reply:
A correction to my previous statement. These instance metrics (with the host and port) are not created for other database operations because we don't have access to the host and port at those times. So these instance metrics will ONLY be the short-lived metrics created when adding a server, and the longer database metrics will be separated under operation metrics.

Reply:
This thread seems obsolete after 3ed5bc54bb0d90bc792ed32cd650cfb87c7de972.